test/test_sparse.py

```diff
@@ -823,6 +823,69 @@ def test_shape(sparse_dims, nnz, with_size):
         test_shape(4, 10, [100, 100, 100, 5, 5, 5, 0])
         test_shape(4, 0, [0, 0, 100, 5, 5, 5, 0])
 
+    @skipIfRocm
+    def test_sparse_sum(self):
+        nnz = 10
+        sparse_dims = 2
+        with_size = [5, 5, 3, 4]
+        S, _, _ = self._gen_sparse(sparse_dims, nnz, with_size)
+        S = S.coalesce()
+        D = S.to_dense()
+
+        self.assertEqual(torch.sparse.sum(S), D.sum())
+
+        # raise at SparseTensor.sum()
+        self.assertRaises(RuntimeError, lambda: S.sum())
+
+        test_dims = []
+        for i in range(1, 5):
+            test_dims += itertools.combinations(range(len(with_size)), i)
+
+        keepdim = [False, True]
+
+        for td in test_dims:
+            for k in keepdim:
+                S_copy = S.clone()
+                S_sum = torch.sparse.sum(S, td, k)
+                D_sum = D.sum(td, k)
+                self.assertEqual(S_sum.to_dense() if S_sum.is_sparse else S_sum, D_sum)
+                self.assertEqual(S_copy.to_dense(), S.to_dense())  # ensure S is not changed
+
+        # raise at dims out of range
+        self.assertRaises(RuntimeError, lambda: torch.sparse.sum(S, 5))
+
+        # test backward
```

Comment:
You are kind of trailblazing here, since historically we didn't have any backwards for sparse. It would be really good to get gradcheck working in the sparse case. Doesn't have to be this PR, but gradcheck tests will give much more assurance than manually testing at a few points. (In this case, sum is pretty simple to check, but this won't always be the case.)

Reply:
Thanks! Yeah I will put gradcheck to my TODO list.